### ISSUE TITLE
"Unwanted Cargo" believes you have a translator device slightly too early

### ISSUE THREAD

**Issue by SolraBizna** [endless-sky/endless-sky#4492]
**Describe the bug**
The "Unwanted Cargo" branch tests "Human Cultural Archives: done", which seems to get set after you land on Alexandria, not after you return to Greenwater.

**To Reproduce**
(Without accidentally triggering Unwanted Cargo...)
1. Proceed in the Wanderer missions until Eruk asks you to get the "human cultural data" from Alexandria.
2. Land on Hai planets until the "hai in cargo" event appears and ticks down.
3. Land on Alexandria with at least 40 credits in the bank.

**Expected behavior**
You encounter the Hai baby and it jabbers unintelligibly.

**Actual behavior**
You encounter the Hai baby and it asks for its mother via your translator device.

**System (please complete the following information):**
 - OS: Debian GNU/Linux
 - Game Source: self-compiled
 - Version: "commit hash 9c8f83c9 :)"

**Additional context**
It's really hard to trigger this except on purpose.

**Comment by Amazinite** [comment]
Good catch. Fixed in https://github.com/endless-sky/endless-sky/commit/da1322aca72d2f30f890ba5527ce9f45f0634e47. This fix will also make it so that if the player ever loses the translator for whatever reason (i.e. they lose the `"language: Wanderer"` condition), this mission won't assume that the player still has it just because a certain mission was offered/done.
